### ISSUE TITLE
Development status

### ISSUE THREAD

**Issue by J3r3Xy** [rpeshkov/IntelWifi#15]
Is the development still going? I'm interested in helping as much as I can (although I don't have much MacOS programming experience)

**Comment by JustusFluegel** [comment]
yeah I can help too if my help is wanted. If nobody is working on this, I may try to develop my own one based on this if this is ok. But I will have to wait until I have enough money to buy me an Xiaomi mi notebook air 13.3 :D

**Comment by rpeshkov** [comment]
Hi guys,

Development status can be described as: freezed until I find enough time on it. 

If someone wants to try continuing this project - feel free to fork it and continue. However, I want to warn you about the amount of work need to be done in order to make this project fully functional: it’s muuuuch more than you might think. It’s not enough to just port iwlwifi code from Linux. 80211 stack of macOS is closed, so you’ll need to deal with that somehow. I previously started another project (black80211) about digging deep into 80211 stack of macOS, but this project only scratches the surface. Yes, there is implementation of some things like displaying wireless network in the list, enabling and disabling WiFi, etc., but you’ll need to implement many other things by yourself, like decrypting frames in case of secured network, parsing beacons and so on.

I really hope this project will be finished one day, but that’s really huge amount of work.

**Comment by MuntashirAkon** [comment]
Using the private framework (ie. Apple80211.framework) is a sad business. It worked in the past, then Apple made a lot of change.

In the black80211 project, you've used the open-sourced IONetworkingFamily, I see.

**Comment by MuntashirAkon** [comment]
I think writing a driver for wpa_supplicant (and a kext for detecting the card) might work. Realtek's USB Wireless devices work this way. They have a kext called `RtWlanU` which detects the device and they use a (sadly, private) driver for wpa_supplicant to control it. Apple will not detect it as a wireless driver, but wifi would at least work.

(BTW, I don't know if its possible, its just an idea)
